vocode/streaming/synthesizer/eleven_labs_synthesizer.py

```diff
@@ -69,36 +80,64 @@ async def create_speech(
-                    f"synthesizer.{SynthesizerType.ELEVEN_LABS.value.split('_', 1)[-1]}.convert",
-                )
-                audio_segment: AudioSegment = AudioSegment.from_mp3(
-                    io.BytesIO(audio_data)  # type: ignore
-                )
-
-                output_bytes_io = io.BytesIO()
-
-                audio_segment.export(output_bytes_io, format="wav")  # type: ignore
-
-                result = self.create_synthesis_result_from_wav(
-                    file=output_bytes_io,
-                    message=message,
-                    chunk_size=chunk_size,
-                )
-                convert_span.end()
-
-                return result
+
+        session = aiohttp.ClientSession()
+
+        response = await session.request(
+            "POST",
+            url,
+            json=body,
+            headers=headers,
+            timeout=aiohttp.ClientTimeout(total=15),
+        )
+
+        if not response.ok:
+            raise Exception(f"ElevenLabs API returned {response.status} status code")
+
+        async def output_generator(
+            response, session
+        ) -> AsyncGenerator[SynthesisResult.ChunkResult, None]:
+            stream_reader = response.content
+            buffer = bytearray()
+
+            # Create a task to send the mp3 chunks to the PydubWorker's input queue in a separate loop
+            async def send_chunks():
+                async for chunk in stream_reader.iter_any():
+                    at_eof = stream_reader.at_eof()
+                    # Send the mp3 chunk and the flag to the PydubWorker's input queue
+                    self.pydub_worker.consume_nonblocking((chunk, at_eof))
+                    # If this is the last chunk, break the loop
+                    if at_eof:
+                        break
+
+            send_task = asyncio.create_task(send_chunks())
+
+            # Await the output queue of the PydubWorker and yield the wav chunks in another loop
+            while True:
+                # Get the wav chunk and the flag from the output queue of the PydubWorker
+                wav_chunk, is_last = await self.pydub_worker.output_queue.get()
+
+                buffer.extend(wav_chunk)
+
+                if len(buffer) >= chunk_size or is_last:
+                    if is_last:
+                        await session.close()
+                    yield SynthesisResult.ChunkResult(buffer, is_last)
+                    buffer.clear()
+                # If this is the last chunk, break the loop
+                if is_last:
+                    create_speech_span.end()
+                    break
+
+            # Wait for the send task to finish and close the session
+            await asyncio.gather(
+                send_task,
+                session.close(),
+            )
+
+        
+
+        return SynthesisResult(
+            output_generator(response, session),  # should be wav
+            lambda _: "",  # useless for now
```

Comment:
I think we have some estimation code for this: https://github.com/vocodedev/vocode-python/blob/4ce40fcfae6543d904f9b78b81f32aa77b3d9df0/vocode/streaming/synthesizer/base_synthesizer.py#L182-L188

let's use this for now and set some reasonable WPM